### PR TITLE
bug(test): Boto3 test setup fix

### PR DESCRIPTION
**Pull request by mmanciop**
By mistake we have been testing the botocore instrumentation (which works with boto3) and thought we were qualifying boto2.

This change adjusts the setup so that we test and report compatibility on the right packages.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -140,7 +140,7 @@ def integration_tests_boto3(
     session,
     boto3_version,
 ):
-    with TestedVersions.save_tests_result("boto3", "boto3", boto3_version):
+    with TestedVersions.save_tests_result("botocore", "boto3", boto3_version):
         install_package("boto3", boto3_version, session)
```
Comment (GuyMoses) [inline noxfile.py:144]:
```suggestion
        install_package("botocore", botocore_version, session)
```

File: noxfile.py [hunk 2]
```diff
@@ -140,7 +140,7 @@ def integration_tests_boto3(
     session,
     boto3_version,
 ):
-    with TestedVersions.save_tests_result("boto3", "boto3", boto3_version):
+    with TestedVersions.save_tests_result("botocore", "boto3", boto3_version):
```
Comment (GuyMoses) [inline noxfile.py:143]:
```suggestion
    with TestedVersions.save_tests_result("botocore", "botocore", botocore_version):
```

**Review by GuyMoses — CHANGES_REQUESTED** [review]

File: noxfile.py (hunk 1 above)
Comment (mmanciop) [reply noxfile.py:144]: We are actually testing boto3 here, even if the instrumentation sits at botocore level. We should express our dependencies in terms of what the users are likely to import.

File: noxfile.py (hunk 2 above)
Comment (mmanciop) [reply noxfile.py:143]: Sane comment as above

**Review by GuyMoses — APPROVED** [review]

**Pull request merged**